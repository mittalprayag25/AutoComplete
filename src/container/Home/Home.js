import React, { useState } from 'react';
import AutoComplete from "../../component/autocomplete/AutoComplete";
import './Home.css';

const App = () => {
  const [options, setOptions] = useState(['India', 'Singapore', 'China', 'America', 'Australia', 'Kenya', 'South Afirca', 'Iraq', 'Canada']);
  const [value, setValue] = useState(null);
  const [selectedOptions, setSelectedOptions] = useState([]);

  const onChange = (data) => {
    const arr = [...selectedOptions, ...data];
    setSelectedOptions(arr.filter(unique));
  };

  const unique = (value, index, self) => {
    return self.indexOf(value) === index;
  };

  const onRemove = (item) => {
    const arr = arrayRemoveElement([...selectedOptions], item);
    setSelectedOptions(arr.filter(unique));
  };

  const arrayRemoveElement = (arr, item) => {
    return arr.filter((ele) => {
      return ele !== item;
    })
  }

  return (
    <div className={"homeContainer"}>
      <AutoComplete
        value={value}
        data-testid="autocomplete"
        options={options}
        onChange={onChange}
        multiValues={selectedOptions}
        onRemove={onRemove}
        onTextChange={setValue} />
    </div>
  );
}
export default App;