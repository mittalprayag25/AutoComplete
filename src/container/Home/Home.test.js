import React from 'react';
import { render } from '@testing-library/react';
import Home from './Home';

test('renders Home', () => {
  const { asFragment } = render(<Home />);
  expect(asFragment()).toMatchSnapshot();
});
