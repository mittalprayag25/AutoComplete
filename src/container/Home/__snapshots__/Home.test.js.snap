// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home 1`] = `
<DocumentFragment>
  <div
    class="homeContainer"
  >
    <div
      class="mainContainer"
    >
      <div
        class="searchableContainer"
      >
        <div
          class="multiSelectContainer"
        >
          <input
            class="searchInput"
            data-testid="searchInput"
            placeholder="Search.."
            type="text"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
